tests/test_explanation.py

```diff
@@ -58,3 +58,45 @@ def test_explanation_hstack_errors(random_seed):
             base_values=np.ones(20) * 0.987,
         )
         _ = base_exp.hstack(exp2)
+
+
+def test_square_explanation_feature_names_slicing_issue_2699(random_seed):
```

Comment:
Nit: perhaps consider a test name that describes the functionality rather than the GH issue, e.g.
`test_explanation_slicing_for_square_arrays`